src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 13;
+        result = 31 * result + (width == null ? 0 : width.hashCode());
+        result = 31 * result + (length == null ? 0 : length.hashCode());
+        result = 31 * result + (color == null ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle current = (Rectangle) rectangle;
+            return (color == current.color && length == current.length
+                    && width == current.width) || (color.equals(current.color)
+                    && length.equals(current.length) && width.equals(current.width));
```

Comment:
let’s follow the following logical flow for each field:

return true,
if both `field` and `rectangle.field` are null
OR if `field` is not null and equals `rectangle.field`